File: Antibody.CareToKnowPro.CRM/Models/EventEntity.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace Antibody.CareToKnowPro.CRM.Models
{
    public partial class EventEntity
    {
        public EventEntity()
        {
            EventEntityProperty = new HashSet<EventEntityProperty>();
        }

        public long EventEntityId { get; set; }
        public long EventId { get; set; }
        public int ActionType { get; set; }
        public int Position { get; set; }
        public int UserId { get; set; }

        [ForeignKey("EventId")]
        [InverseProperty("EventEntity")]
        public virtual Event Event { get; set; }
        [ForeignKey("UserId")]
        [InverseProperty("EventEntity")]
        public virtual User User { get; set; }
        [InverseProperty("EventEntity")]
        public virtual ICollection<EventEntityProperty> EventEntityProperty { get; set; }
    }
}

File: Antibody.CareToKnowPro.CRM/ClientApp/src/app/model/eventEntity.ts

import { IEvent as Event } from './event';
import { IEventEntityProperty as EventEntityProperty } from './eventEntityProperty';
import { IUser as User } from './user';

export interface IEventEntity { 
    eventEntityId?: number;
    eventId?: number;
    actionType?: number;
    position?: number;
    userId?: number;
    event?: Event;
    user?: User;
    eventEntityProperty?: Array<EventEntityProperty>;
}
